### PR TITLE
Refine the implementation of rename

### PR DESCRIPTION
**Pull request by xiaoxiang781216**
## Summary
Found in reviewing https://github.com/apache/incubator-nuttx/pull/7312

- fs/vfs: Free subdir before allocate new one in pseudorename
- fs/vfs: Remove the redundancy file name comparison in mountptrename 

## Impact
Minor

### CODE REVIEW

File: fs/vfs/fs_rename.c [hunk 1]
```diff
@@ -90,7 +90,7 @@ static int pseudorename(FAR const char *oldpath, FAR struct inode *oldinode,
         {
           inode_release(newinode);
           ret = OK;
-          goto errout; /* Bad naming, this is not an error case. */
+          goto errout; /* Same name, this is not an error case. */
```
Comment (acassis) [inline fs/vfs/fs_rename.c:93]: Should it be considered an error? At least on Linux when trying to move a directory to itself we see an error reported:
```
$ mv a a
mv: cannot move 'a' to a subdirectory of itself, 'a/a'
```

Comment (xiaoxiang781216) [reply fs/vfs/fs_rename.c:93]: @acassis the message is reported by mv command, not function Here is my test on WSL2:
```
$ cat test.c
#include <stdio.h>

int main(int argc, char *argv[])
{
    return rename(argv[1], argv[2]);
}
$ gcc test.c 
$ ./a.out a.out a.out 
$ echo $?
0
```

File: fs/vfs/fs_rename.c [hunk 2]
```diff
@@ -111,26 +111,21 @@ static int pseudorename(FAR const char *oldpath, FAR struct inode *oldinode,
...
         {
           FAR char *subdirname;
-          FAR char *tmp;
 
           inode_release(newinode);
+          if (subdir != NULL)
+            {
+              kmm_free(subdir);
+              subdir = NULL;
+            }
```
[the remark this answers is not in the document]
Comment (xiaoxiang781216) [reply fs/vfs/fs_rename.c:123]: Done.

File: fs/vfs/fs_rename.c (hunk 1 above)
Comment (pkarashchenko) [reply fs/vfs/fs_rename.c:93]: Yeah. Probably will need to update `cmd_mv` in NSH

**Review by pkarashchenko — APPROVED** [review]

File: fs/vfs/fs_rename.c (hunk 1 above)
Comment (xiaoxiang781216) [reply fs/vfs/fs_rename.c:93]: Could be

**Review by pkarashchenko — APPROVED** [review]

**Review by acassis — APPROVED** [review]

**Pull request merged**